### ISSUE TITLE
Changes to “changed solute” alerts

### ISSUE THREAD

**Issue by Matthew-phet** [phetsims/molarity#127]
**Values Checked**

- **Unsaturated to unsaturated** (e.g., Drink mix to Potassium chromate):
   - “{{Yellow}} solution, concentration {{1.000 molar}}.” 
- **Unsaturated to saturated** (e.g. Potassium chromate to Potassium permanganate):
   - “{{Purple}} solution, saturated with {{a few solids}}, concentration {{0.500 molar}}
- **Saturated to saturated** (Potassium permanganate to Potassium dichromate):
   - “{{Orange}} solution, saturated with {{a few solids}}, concentration {{0.500 molar}}
- **Saturated to unsaturated** (Potassium dichromate to Gold(III) chloride):
   - “{{Gold}} solution, concentration {{1.000 molar}}.”

**Values not checked**

- **Unsaturated to unsaturated** (e.g., Drink mix to Potassium chromate):
   - “{{Yellow}} solution, {{slightly concentrated}}.” 
- **Unsaturated to saturated** (e.g. Potassium chromate to Potassium permanganate):
   - “{{Purple}} solution, saturated with {{a few solids}}, {{beyond max concentration}}.
- **Saturated to saturated** (Potassium permanganate to Potassium dichromate):
    - “{{Orange}} solution, saturated with {{a few solids}}, {{beyond max concentration}}.
- **Saturated to unsaturated** (Potassium dichromate to Gold(III) chloride):
   - “{{Gold}} solution, {{slightly concentrated}}.”

**Edge Case, change of solute when there is no solute in beaker**
**Values not checked**

e.g. Drink mix to Potassium permanganate:

- {{Clear}} solution, pure water with no solute.

**Values checked**
e.g. Drink mix to Potassium permanganate

- {{Clear}} solution, pure water at 0.000 molar.

**OLD VERSION**
When solute is changed from ‘not saturated’ solution to a ‘not saturated’ solution: 
- New solution. {{Drink Mix}} concentration readout range {{0.00}} to {{5.0 M}}

From a ‘saturated’ solution to a ‘not saturated’ solution:
- New solution. Not saturated, no solids. Solution {{very concentrated}}. 

From a ‘not saturated’ or from a ‘saturated’ solution to a ‘saturated’ solution:
- New solution. Saturated, {{a lot of}} solids. Beyond max concentration.

**Comment by twant** [comment]
@terracoda @Matthew-phet this is mostly implemented. Just need to add the edge cases and fix capitalization. not assigning over to you yet, but feel free to take a look so far and let me know if you see any other improvements on this one.

**Comment by terracoda** [comment]
@twant, the alerts (i.e., context responses) sound great. 
I'd like to have another listen once the concentration levels from #117 are fully implemented.
Nice work!

**Comment by terracoda** [comment]
Please assign back to @Matthew-phet and @terracoda once the concentration levels (#117) are implmented.

**Comment by terracoda** [comment]
@twant, apologies we should have put priorities in the issues, so  just marking as blocked until #117 is done. Please re-assign and re-label at that point.

**Comment by twant** [comment]
@terracoda @Matthew-phet assigning back over for review now that #117 has been implemented.

**Comment by terracoda** [comment]
@twant, I made some comments in #117 that might be relevant to this issue.

**Comment by terracoda** [comment]
Here are some issues with the alerts:
**Solution Values Checked:**
<img width="307" alt="Screen Shot 2019-07-23 at 3 44 52 PM" src="https://user-images.githubusercontent.com/1902116/61717671-d0525380-ad61-11e9-8ece-e7962a741477.png">

**Solution Values not checked:**
The wording of the alerts look good as long as the qualitative concentration level is correct. 
<img width="307" alt="Screen Shot 2019-07-23 at 3 45 33 PM" src="https://user-images.githubusercontent.com/1902116/61718200-c715b680-ad62-11e9-9185-98f42a9204ab.png">

**Comment by terracoda** [comment]
@twant, I just want to re-check concentration levels.

**Comment by terracoda** [comment]
I'll check the edge case next...

**Comment by terracoda** [comment]
@twant, in the edge case (Solution Values checked) I made a mistake in the example I provided. It's supposed to say moles, not molar! Apologies.

The alert should cue users to go play with the Solute Amount slider.
In the Solution Values not checked we say, "Clear solution, pure water with no solute."
So with Solution Values checked we should say, "Clear solution, pure water at 0.000 moles."

Though, I would be fine with:
- "Clear solution, pure water at 0 moles."
- "Clear solution, pure water at zero moles."

Saying all the zeros may be accurate but not that helpful

**Comment by terracoda** [comment]
@twant, I didn't mention capitalization, as I just want to get words right first. In alerts, however, I find it is safe to use proper sentence capitalization and punctuation. If all `aria-valuetext` parameters are delivered in lower case, and these parameters do not actually start alert phrases then we have fewer places to worry about in terms of getting capitalization right in alerts.

Anyways, in this issue, we can just focus on the words.

**Comment by twant** [comment]
@terracoda @Matthew-phet pushed changes to both wording and capitalization -- will be in the dev version tonight. Assigning back over to you for review!

**Comment by terracoda** [comment]
I am marking this issue for our Tuesday design meeting. There are still some issues. I'll try to get them documented before Tuesday.

**Comment by terracoda** [comment]
I still need to examine the screen shots more carefully and explore edge cases, but I think we are getting qualitative descriptions in the quantitative alerts in some cases which might be incorrect.
<img width="273" alt="127_values_checked_issues_remain" src="https://user-images.githubusercontent.com/1902116/62802830-b8870900-ba9d-11e9-9e43-30b66789429a.png">
<img width="291" alt="127_values_hidden_looks_good" src="https://user-images.githubusercontent.com/1902116/62802846-bfae1700-ba9d-11e9-8948-54b38ee16456.png">

**Comment by terracoda** [comment]
@Matthew-phet, @twant, I will review this issue more before Tuesday, but I just ran out of time today.

Feel free to examine it yourselves, though.

**Comment by terracoda** [comment]
**Solution Values checked - quantitative experience**
Alerts have changed, but are still not quite right.
<img width="447" alt="Screen Shot 2019-08-11 at 3 05 18 PM" src="https://user-images.githubusercontent.com/1902116/62840123-50c1f100-bc4a-11e9-9332-2f3a87be35e5.png">

**Solution Values not checked - qualitative experience**
Alerts are still correct.
<img width="382" alt="Screen Shot 2019-08-11 at 3 09 48 PM" src="https://user-images.githubusercontent.com/1902116/62840110-25d79d00-bc4a-11e9-9915-2b4ec76d191a.png">

**I think the issue is:**
If the solution is saturated, we get a described concentration value instead of the concentration value in molar. 
I think what we want (with solution values checked) is the concentration value in raw molars rather than in the described qualitative value, "concentration {{max concentration}}".
So instead of:
- Orange solution, saturated with a few solids, concentration max concentration
- Purple solution, saturated with a few solids, concentration max concentration
We want:
- Orange solution, saturated with a few solids, concentration 0.500 molar
- Purple solution, saturated with a few solids, concentration 0.500 molar

These two solutions have the same saturation point, but other solutions have different ones, for example with different Solute/Volume values a change to saturated Copper (II) sulphate would be:
- Blue solution, saturated with a few solids, concentration 1.400 molar

@Matthew-phet and @emily-phet would you agree?

**Comment by terracoda** [comment]
We can discuss this on Tuesday if more discussion is needed.

**Comment by terracoda** [comment]
@Matthew-phet, we didn't get to this issue on Tuesday, but if you feel the examples in https://github.com/phetsims/molarity/issues/127#issuecomment-520266202 are correct, can you plesae assign to @twant and remove the meeting label?

Assigning to you for now.

**Comment by twant** [comment]
Ok, I've updated this one! Assigning back over

**Comment by twant** [comment]
@Matthew-phet and I checked this one today and agreed it's fixed. Closing now!
